{
  "cve": "CVE-2013-4314",
  "mitre": {
    "cpes": [],
    "created": "2013-09-30T21:00:00+00:00",
    "description": "The X509Extension in pyOpenSSL before 0.13.1 does not properly handle a '\\0' character in a domain name in the Subject Alternative Name field of an X.509 certificate, which allows man-in-the-middle attackers to spoof arbitrary SSL servers via a crafted certificate issued by a legitimate Certification Authority.",
    "metrics": {
      "cvssV2_0": {},
      "cvssV3_0": {},
      "cvssV3_1": {},
      "cvssV4_0": {}
    },
    "mitre_repo_path": "cves/2013/4xxx/CVE-2013-4314.json",
    "references": [
      "http://lists.opensuse.org/opensuse-updates/2013-11/msg00015.html",
      "http://www.debian.org/security/2013/dsa-2763",
      "http://www.openwall.com/lists/oss-security/2013/09/06/2",
      "http://www.ubuntu.com/usn/USN-1965-1",
      "https://bugzilla.redhat.com/show_bug.cgi?id=1005325",
      "https://mail.python.org/pipermail/pyopenssl-users/2013-September/000478.html"
    ],
    "title": null,
    "updated": "2013-12-05T17:57:00+00:00",
    "vendors": [],
    "weaknesses": []
  },
  "nvd": {
    "cpes": [
      "cpe:2.3:a:jean-paul_calderone:pyopenssl:*:*:*:*:*:*:*:*",
      "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.10:*:*:*:*:*:*:*",
      "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.11:*:*:*:*:*:*:*",
      "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.11:a1:*:*:*:*:*:*",
      "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.11:a2:*:*:*:*:*:*",
      "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.12:*:*:*:*:*:*:*",
      "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.7:*:*:*:*:*:*:*",
      "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.8:a1:*:*:*:*:*:*",
      "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.9:*:*:*:*:*:*:*",
      "cpe:2.3:o:canonical:ubuntu_linux:10.04:-:lts:*:*:*:*:*",
      "cpe:2.3:o:canonical:ubuntu_linux:12.04:-:lts:*:*:*:*:*",
      "cpe:2.3:o:canonical:ubuntu_linux:12.10:*:*:*:*:*:*:*",
      "cpe:2.3:o:canonical:ubuntu_linux:13.04:*:*:*:*:*:*:*"
    ],
    "created": "2013-09-30T21:55:09.583000+00:00",
    "description": "The X509Extension in pyOpenSSL before 0.13.1 does not properly handle a '\\0' character in a domain name in the Subject Alternative Name field of an X.509 certificate, which allows man-in-the-middle attackers to spoof arbitrary SSL servers via a crafted certificate issued by a legitimate Certification Authority.",
    "metrics": {
      "cvssV2_0": {
        "score": 4.3,
        "vector": "AV:N/AC:M/Au:N/C:N/I:P/A:N"
      },
      "cvssV3_0": {},
      "cvssV3_1": {},
      "cvssV4_0": {}
    },
    "nvd_repo_path": "2013/CVE-2013-4314.json",
    "references": [
      "http://lists.opensuse.org/opensuse-updates/2013-11/msg00015.html",
      "http://www.debian.org/security/2013/dsa-2763",
      "http://www.openwall.com/lists/oss-security/2013/09/06/2",
      "http://www.ubuntu.com/usn/USN-1965-1",
      "https://bugzilla.redhat.com/show_bug.cgi?id=1005325",
      "https://mail.python.org/pipermail/pyopenssl-users/2013-September/000478.html"
    ],
    "title": null,
    "updated": "2013-12-08T06:00:25.607000+00:00",
    "vendors": [
      "canonical",
      "canonical$PRODUCT$ubuntu_linux",
      "jean-paul_calderone",
      "jean-paul_calderone$PRODUCT$pyopenssl"
    ],
    "weaknesses": [
      "CWE-20"
    ]
  },
  "opencve": {
    "changes": [],
    "cpes": {
      "data": [
        "cpe:2.3:a:jean-paul_calderone:pyopenssl:*:*:*:*:*:*:*:*",
        "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.10:*:*:*:*:*:*:*",
        "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.11:*:*:*:*:*:*:*",
        "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.11:a1:*:*:*:*:*:*",
        "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.11:a2:*:*:*:*:*:*",
        "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.12:*:*:*:*:*:*:*",
        "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.7:*:*:*:*:*:*:*",
        "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.8:a1:*:*:*:*:*:*",
        "cpe:2.3:a:jean-paul_calderone:pyopenssl:0.9:*:*:*:*:*:*:*",
        "cpe:2.3:o:canonical:ubuntu_linux:10.04:-:lts:*:*:*:*:*",
        "cpe:2.3:o:canonical:ubuntu_linux:12.04:-:lts:*:*:*:*:*",
        "cpe:2.3:o:canonical:ubuntu_linux:12.10:*:*:*:*:*:*:*",
        "cpe:2.3:o:canonical:ubuntu_linux:13.04:*:*:*:*:*:*:*"
      ],
      "providers": [
        "nvd"
      ]
    },
    "created": {
      "data": "2013-09-04T00:00:00+00:00",
      "provider": "redhat"
    },
    "description": {
      "data": "The X509Extension in pyOpenSSL before 0.13.1 does not properly handle a '\\0' character in a domain name in the Subject Alternative Name field of an X.509 certificate, which allows man-in-the-middle attackers to spoof arbitrary SSL servers via a crafted certificate issued by a legitimate Certification Authority.",
      "provider": "mitre"
    },
    "metrics": {
      "cvssV2_0": {
        "data": {
          "score": 4.3,
          "vector": "AV:N/AC:M/Au:N/C:N/I:P/A:N"
        },
        "provider": "nvd"
      },
      "cvssV3_0": {
        "data": {},
        "provider": null
      },
      "cvssV3_1": {
        "data": {},
        "provider": null
      },
      "cvssV4_0": {
        "data": {},
        "provider": null
      },
      "kev": {
        "data": {},
        "provider": null
      },
      "ssvc": {
        "data": {},
        "provider": null
      },
      "threat_severity": {
        "data": "Moderate",
        "provider": "redhat"
      }
    },
    "references": {
      "data": [
        "http://lists.opensuse.org/opensuse-updates/2013-11/msg00015.html",
        "http://www.debian.org/security/2013/dsa-2763",
        "http://www.openwall.com/lists/oss-security/2013/09/06/2",
        "http://www.ubuntu.com/usn/USN-1965-1",
        "https://bugzilla.redhat.com/show_bug.cgi?id=1005325",
        "https://mail.python.org/pipermail/pyopenssl-users/2013-September/000478.html",
        "https://nvd.nist.gov/vuln/detail/CVE-2013-4314",
        "https://www.cve.org/CVERecord?id=CVE-2013-4314"
      ],
      "providers": [
        "mitre",
        "nvd",
        "redhat"
      ]
    },
    "title": {
      "data": "pyOpenSSL: hostname check bypassing vulnerability",
      "provider": "nvd"
    },
    "updated": {
      "data": "2013-12-08T06:00:25.607000+00:00",
      "provider": "nvd"
    },
    "vendors": {
      "data": [
        "canonical",
        "canonical$PRODUCT$ubuntu_linux",
        "jean-paul_calderone",
        "jean-paul_calderone$PRODUCT$pyopenssl"
      ],
      "providers": [
        "nvd"
      ]
    },
    "weaknesses": {
      "data": [
        "CWE-20"
      ],
      "providers": [
        "nvd"
      ]
    }
  },
  "redhat": {
    "cpes": [],
    "created": "2013-09-04T00:00:00+00:00",
    "description": "The X509Extension in pyOpenSSL before 0.13.1 does not properly handle a '\\0' character in a domain name in the Subject Alternative Name field of an X.509 certificate, which allows man-in-the-middle attackers to spoof arbitrary SSL servers via a crafted certificate issued by a legitimate Certification Authority.",
    "metrics": {
      "cvssV2_0": {
        "score": 4.3,
        "vector": "AV:N/AC:M/Au:N/C:N/I:P/A:N"
      },
      "cvssV3_0": {},
      "cvssV3_1": {},
      "threat_severity": "Moderate"
    },
    "redhat_repo_path": "2013/CVE-2013-4314.json",
    "references": [
      "https://nvd.nist.gov/vuln/detail/CVE-2013-4314",
      "https://www.cve.org/CVERecord?id=CVE-2013-4314"
    ],
    "title": "pyOpenSSL: hostname check bypassing vulnerability",
    "updated": "2013-09-04T00:00:00+00:00",
    "vendors": [],
    "weaknesses": []
  }
}